docs/integration-guide.md

```diff
@@ -107,6 +107,342 @@ Here's how we recommend doing it:
+`client`. You may also want to define additional parameters using the `REQ`
+object.
+
+In the example above, we have defined `payload` which is populated from the
+body of the http request, `stream` with a default of `test` (available by
+default in Zulip dev environment), and `topic` with a default of `Hello World`.
+
+In the body of the function we define the body of the message as `Hello! I am
+happy to be here! :smile: `. The `:smile:` indicates an emoji. Then we append a
+link to the Wikipedia article of the day as provided by the json payload. If
+the json payload does not include data for `featured_title` and `featured_url`
+we catch a `KeyError` and use `json_error` to return the appropriate
+information: a 400 http status code with relevant details.
+
+Then we send a public (stream) message with `check_send_message`.  Finally, we
+return `json_success()`.
+
+#### Step 2: Create an api endpoint for the webhook
+
+In order for a webhook to be externally available, it must be mapped to a url.
+This is done in `zproject/urls.py`. Look for the lines:
+
+```
+# Incoming webhook URLs
+urls += [
+    # Sorted integration-specific webhook callbacks.
+```
+
+And you'll find the entry for Hello World:
+
+```
+url(r'^api/v1/external/helloworld$',  'zerver.views.webhooks.helloworld.api_helloworld_webhook'),
+```
+
+This tells the Zulip api to call the `api_helloworld_webhook` function in
+`zerver/views/webhooks/helloworld.py` when it receives a request at
+`/api/v1/external/helloworld`.
+
+At this point, if you're following along and/or writing your own Hello World
+webhook, you have written enough code to test your integration.
+
+You can do so by using Zulip itself or curl on the command line.
+
+Using `manage.py` from within Zulip Dev environment:
+
+```
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:/srv/zulip$
+./manage.py send_webhook_fixture_message \
+> --fixture=zerver/fixtures/helloworld/helloworld_hello.json \
+> '--url=http://localhost:9991/api/v1/external/helloworld?api_key=<api_key>'
+```
+After which you should see something similar to:
+
+```
+2016-07-07 15:06:59,187 INFO     127.0.0.1       POST    200 143ms (mem: 6ms/13) (md: 43ms/1) (db: 20ms/9q) (+start: 147ms) /api/v1/external/helloworld (helloworld-bot@zulip.com via ZulipHelloWorldWebhook)
+```
+
+Using curl:
+
+```
+curl -X POST -H "Content-Type: application/json" -d '{ "featured_title":"Marilyn Monroe", "featured_url":"https://en.wikipedia.org/wiki/Marilyn_Monroe" }' http://localhost:9991/api/v1/external/helloworld\?api_key\=<api_key>
+```
+
+After which you should see:
+```
+{"msg":"","result":"success"}
+```
+
+Using either method will create a message in Zulip:
+
+![Image of Hello World webhook message](images/helloworld-webhook.png)
+
+Another helpful tool for testing your integration is to use
+[UltraHook](http://www.ultrahook.com/), which allows you to receive webhook
+calls via your local Zulip dev environment.
```

Comment:
We should explain why this is useful -- e.g. if the webhook sends data using special HTTP headers (bitbucket2 is an example), this can be useful for doing an end-to-end test to figure out what's missing.